### PR TITLE
Make CLI args named and add animation

### LINKED ISSUE #103: Make CLI args named
Since all args are currently optional, you should be able to provide the ones you want only. Currently, to specify the branch name, you need to also specify all other args.

### PR DESCRIPTION
**Pull request by joglr**
Fixes #103

### CODE REVIEW

File: parser/src/index.ts [hunk 1]
```diff
@@ -1,29 +1,122 @@
 import "dotenv/config"
+import { createSpinner } from "nanospinner"
+import { resolve } from "path"
+import { performance } from "perf_hooks"
+import parser from "yargs-parser"
```
Comment (tjomson) [inline parser/src/index.ts:5]: Could we maybe call this arg-parser or something. There are a alot of things called "parser" :D

File: parser/src/index.ts [hunk 2]
```diff
@@ -1,29 +1,122 @@
...
+  job: () => Promise<T>,
+  beforeMsg: string,
+  afterMsg: string,
+  errorMsg: string
+) {
 
+    let success = (text: string, final = false) => {
+      if (getLogLevel() === LOG_LEVEL.SILENT) return
+      if (spinner === null) return log.info(text)
+      // if (final) spinner.stop()
```
Comment (tjomson) [inline parser/src/index.ts:66]: Commented out code?

File: parser/src/index.ts [hunk 3]
```diff
@@ -1,29 +1,122 @@
...
+const start = performance.now()
+const [branchHead, branchName] = await describeAsyncJob(
+  () => findBranchHead(repoDir, branch),
+  "Finding branch head",
+  "Found branch head",
+  "Error finding branch head"
+)
+const outFileName =
+  args.out ?? `./.temp/${getRepoName(repoDir)}_${branchName}.json`
+let repoTree = await describeAsyncJob(
```
Comment (tjomson) [inline parser/src/index.ts:102]: This is cool :O

File: parser/src/index.ts [hunk 4]
```diff
@@ -1,29 +1,122 @@
...
+      success(`${afterMsg} ${suffix}`, true)
+      return result
+    } catch(e) {
+        error(errorMsg)
+        log.error(e)
+        process.exit(1)
+      }
 }
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
```
Comment (tjomson) [inline parser/src/index.ts:88]: Unused var

**Comment by tjomson** [comment]
Can some of the stuff in this commit be refactored to a separate file, so we can keep index clean?

**Comment by joglr** [comment]
> Can some of the stuff in this pr be refactored to a separate file, so we can keep index clean?

Yeah, that's a good idea.

File: parser/src/index.ts (hunk 2 above)
Comment (joglr) [reply parser/src/index.ts:66]: Solved

File: parser/src/index.ts (hunk 4 above)
Comment (joglr) [reply parser/src/index.ts:88]: Removed

**Comment by joglr** [comment]
> > Can some of the stuff in this pr be refactored to a separate file, so we can keep index clean?
> 
> Yeah, that's a good idea.

`index.ts` is now just 4 LoC 🎉

**Review by tjomson — APPROVED** [review]
Epic work 🚀

**Pull request merged**